### PR TITLE
Simplify pandas fixture.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

Followup to #10124.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/testing/conftest.py [hunk 1]
```diff
@@ -80,22 +80,20 @@ def mpl_image_comparison_parameters(request, extension):
...
+    except ImportError:
+        from pandas.tseries.converter import register
+    register()
+    try:
+        yield pd
+    finally:
         try:
             from pandas.plotting import (
                 deregister_matplotlib_converters as deregister)
             request.addfinalizer(deregister)
```
Comment (QuLogic) [inline lib/matplotlib/testing/conftest.py]: Why aren't you just calling it?

Comment (anntzer) [reply lib/matplotlib/testing/conftest.py]: because it was a typo

File: lib/matplotlib/testing/conftest.py [hunk 2]
```diff
@@ -80,22 +80,21 @@ def mpl_image_comparison_parameters(request, extension):
 
 @pytest.fixture
 def pd(request):
```
Comment (tacaswell) [inline lib/matplotlib/testing/conftest.py]: do we still need to take in `request`?

Comment (anntzer) [reply lib/matplotlib/testing/conftest.py]: nope...

**Pull request merged**